packages/donate-button/src/generic-button-loader.tsx

```diff
@@ -0,0 +1,54 @@
+import {render} from 'preact';
+import {
+	GenericButtonProps,
+	DonateButtonOptions
+} from 'src/helpers/options-types';
+
+interface GenericButtonLoader {
+	selector: string;
+	onClick: () => void;
+	widgetOptions: Partial<DonateButtonOptions>;
+	options: Partial<GenericButtonProps>;
+}
+
+const getCharityName = (
+	widgetOptions: Partial<DonateButtonOptions>
+): string => {
+	const genericFoundation = 'your-foundation';
```

Comment:
hm would be nice if we kept this kind of hard-coding all in one place instead of spreading it across different parts of the app

Reply:
Yes, what I realized doing this is that if the user wants to use the `onSubmit` function instead of the params mode we will not have access to the charity name. Probably we should move that property up on the options object.